src/ui/reusable/RealmLoadingComponent/index.tsx

```diff
@@ -198,30 +202,33 @@ export abstract class RealmLoadingComponent<
         // Other errors, propagate it.
         throw error;
       }
-    } else if (realm && realm.mode === realms.RealmLoadingMode.Synced) {
-      const props = (realm as any) as realms.ISyncedRealmToLoad;
+    }
+
+    if (realm && realm.mode === realms.RealmLoadingMode.Synced) {
```

Comment:
I'm not sure this works as expected, but It is unclear to me why not.
I tried to test:
1) Clearing all caches
2) Open the default realm from https://cmelchior.us1.cloud.realm.io. This took ~5 seconds.
3) Close the browser window
4) Open the default Realm again. I would expect this to be instantaneous since the file should be cached, but it it still takes ~5 seconds.

Reply:
Yeah, you'd need to have a huge Realm to notice the difference as the 5 seconds are rendering, logging users in (since we don't serialize right now), etc. If you have a Realm that is several hundred megabytes, then it's quite noticeable.